packages/ag-charts-enterprise/src/gradient-legend/gradientLegend.ts

```diff
@@ -274,29 +277,31 @@ export class GradientLegend {
                 unreachable(this.position);
         }
 
-        switch (this.position) {
-            case 'left':
-            case 'left-top':
-            case 'left-bottom':
-                shrinkRect.shrink(width + this.spacing, 'left');
-                break;
-            case 'right':
-            case 'right-top':
-            case 'right-bottom':
-                shrinkRect.shrink(width + this.spacing, 'right');
-                break;
-            case 'top':
-            case 'top-left':
-            case 'top-right':
-                shrinkRect.shrink(height + this.spacing, 'top');
-                break;
-            case 'bottom':
-            case 'bottom-left':
-            case 'bottom-right':
-                shrinkRect.shrink(height + this.spacing, 'bottom');
-                break;
-            default:
-                unreachable(this.position);
+        if (!this.floating) {
```

Comment:
Nit: Maybe just short-circuit here and remove a nested block with:
```suggestion
        if (this.floating) return { top, left };
```

Reply:
Let's leave this as-is for now. Some additional computation will be needed to add `xOffset` and `yOffset` so `getMeasurement()` will may need to be broken-up to simply it.